src/plugins/customdata.ts

```diff
@@ -0,0 +1,28 @@
+// Allows data sent via. normal DOM event handlers to be logged alongside Clarity data by listening to the
+// custom DOM event "claritydata".
+import { addEvent, bind } from "./../core";
+
+// Events coming from normal DOM will have additional data in details property
+interface ICustomEvent extends Event {
+    detail: any;
+}
+
+export default class CustomData implements IPlugin {
+    private eventName = "Custom";
+
+    public activate() {
+        bind(window, "claritydata", this.customDataHandler.bind(this));
```

Comment:
Added a comment at the top of the file to the DOM documentation for .createEvent() which should help. Also see documentation added in customdata.md.